crypto/encode_decode/decoder_lib.c

```diff
@@ -600,10 +600,10 @@ static int decoder_process(const OSSL_PARAM params[], void *arg)
 
         data->flag_construct_called = 0;
         if (ctx->construct != NULL) {
-            int rv = ctx->construct(decoder_inst, params, ctx->construct_data);
-
+            ERR_set_mark();
+            ok = ctx->construct(decoder_inst, params, ctx->construct_data) > 0;
+            ERR_pop_to_mark();
```

Comment:
This doesn't look right

Reply:
This prevents
```
ossl_store_handle_load_result:crypto/store/store_result.c:151:error: unsupported:
```
If you have a better solution just let me know.

Reply:
Then you should probably modify crypto/store/store_result.c.

Reply:
Why - or where exactly?
I think `ossl_store_handle_load_result()` legitimately throws
```
    if (*v == NULL)
        ERR_raise(ERR_LIB_OSSL_STORE, ERR_R_UNSUPPORTED);
```

Reply:
The new error on `apps/openssl x509 -in test/certs/v3-certs-RC2.p12 -passin 'pass:v3-certs'` pops up after my above change
``` 
            ERR_set_mark();
            ok = ctx->construct(decoder_inst, params, ctx->construct_data) > 0;
            ERR_pop_to_mark();
```
which may indicate that my change was inadequate or that it revealed another issue.

Reply:
Oops, must correct my last statement.
The new error is related to my **other** changes, those to `decode_der2key.c`.

Reply:
Argh, I was confused about the nature of that test, but meanwhile I understand that as I had originally written it is related to the above change to `decoder_lib.c`.
I meanwhile see and agree that the above change cannot be the correct solution 
because it throws away relevant error information regarding `RC2-40-CBC`.